src/components/common/BcRegEntityDetails.vue

```diff
@@ -0,0 +1,310 @@
+<template>
+  <div>
+    <template v-if="isBenefit">
+      <div class="my-6">
+        <p class="subtitle">
+          {{ updatedArticleTitle }}
+        </p>
+        <p class="info-text">
+          Before submitting your alteration notice you <span class="font-weight-bold">must
+            change your company's articles to include a set of {{ updatedArticleTitle }}</span> OR draft new articles
+          containing a benefit provision.
+        </p>
+      </div>
+
+      <section class="text-body-3">
+        <!-- Help Section Toggle -->
+        <HelpSectionToggle
+          :isHelpVisible="helpToggle"
+          @toggle="helpToggle = !helpToggle"
+        />
```

Comment:
This component ^^ doesn't do much. I think it would be a better design to have this component handle the showing and hiding of the content (ie, fully encapsulate the show/hide logic).

You could do this by passing a template with the content to display (ie, the help section below).

See also HelpSection.vue.

Reply:
HelpSection.vue is fairly specific, it probably should have been retooled to support slots

Reply:
Probably. I'm suggesting to keep this one but with slots.

Reply:
I put this section back instead of a component, maybe better not changing HelpSection.vue.

Reply:
I liked the HelpSectionToggle component and recommend you revert it.

My comment above is that you should pass a slot to the component (which is currently your section block), and have all the show/hide logic in the component. Currently that logic is both in BcRegEntityDetails and HelpSectionToggle. You can free BcRegEntityDetails from knowing anything about whether the help is shown/hidden, which is a nice abstraction/ encapsulation/ separation of concern. For example:

In BcRegEntityDetails.vue:
```
<HelpSectionToggle>
  <template #content> <!-- you may be able to use section instead of template here -->
    <!-- your help content goes here -->
  </template>
```

In HelpSectionToggle.vue:
```
<template>
  <div class="help-section-toggle">
    <div
      class="info-text help-toggle"
      @click="isHelpVisible= !isHelpVisible"
    >
      <!-- put your v-icon and show/hide span here -->

      <slot name="content" v-if="isHelpVisible" />
</div>
</template>
```

Reply:
I'm thinking move it to ChangeBusinessType.vue:
```
        <BcRegEntityDetails
          :helpToggle="helpToggle"
          ...
        >
          <template #toggle>
            <div
              class="info-text help-toggle pt-2"
              @click="helpToggle = !helpToggle"
            >
              <v-icon
                class="pr-2 mt-n1"
                color="primary"
              >
                mdi-help-circle-outline
              </v-icon>
              <span v-if="!isHelpVisible">Learn More</span>
              <span v-else>Hide Learn More</span>
            </div>
          </template>
        </BcRegEntityDetails>
```
at BcRegEntityDetails.vue:
```
<!-- Help Section Toggle -->
<slot name="toggle" />
```
feel HelpSectionToggle can be removed cause it's too simple. what do you think?

Reply:
Would you have multiple copies of the above?

I'm looking for simple/clean/easy-to-review code and use of sub-components as needed to accomplish that.

There are different ways to lay this out. Can I leave this with you?

Reply:
no multiple copies, thanks for that! I updated all code under <!-- Help Section Toggle --> on BcRegEntityDetails with slots, I think better to keep as its own template cause it's specific to alteration, not for common components.

Reply:
You've done things differently than I would have, and that's OK.

My initial concern -- and it is still a concern -- is that the visibility variables are in 2 components (parent and child), and I was trying to get away from that. (In fact, you have both helpToggle and isHelpVisible in ChangeBusinessType.vue.)

I think it would be more simple to just have a help component, which handles its own show/hide, that you pass a slot of help data to.

Reply:
I've replaced isHelpVisible with helpToggle. Both of these variables were already present in the existing alteration code, and I was simply utilizing them. I didn't realize initially that they were serving the same purpose. Therefore, I've now removed isHelpVisible.

Now, only ChangeBusinessType.vue is handling the show/hide logic, while BcRegEntityDetails.vue is merely retrieving the display status.

Reply:
Did you try this? It seems simpler to me, and only has visibility/display status in 1 component.

> In BcRegEntityDetails.vue:
> 
> ```
> <HelpSectionToggle>
>   <template #content> <!-- you may be able to use section instead of template here -->
>     <!-- your help content goes here -->
>   </template>
> </HelpSectionToggle>
> ```
> 
> In HelpSectionToggle.vue:
> 
> ```
> <template>
>   <div class="help-section-toggle">
>     <div @click="isHelpVisible= !isHelpVisible">
>       <!-- put your v-icon and show/hide span here -->
>   </div>
> 
>   <slot name="content" v-if="isHelpVisible" />
> </template>
> ```

Reply:
I added back HelpSectionToggle.vue with slots. and only let it control visibility.

Reply:
Sorry for the extra work, but it looks great! And it's clean, compact, reusable, etc. Thanks for fixing it.

Reply:
thanks for point out, it's good opportunity to learn vue slots.